libs/process/opts.go

```diff
@@ -69,10 +70,40 @@ func WithStdoutWriter(dst io.Writer) execOption {
 	}
 }
 
+// safeMultiWriter is a thread-safe io.Writer that writes to multiple writers.
+// It is functionality equivalent to io.MultiWriter, but is safe for concurrent use.
+type safeMultiWriter struct {
+	writers []io.Writer
+	mu      sync.Mutex
+}
+
+// newSafeMultiWriter creates a new safeMultiWriter that writes to the provided writers.
+func newSafeMultiWriter(writers ...io.Writer) *safeMultiWriter {
+	return &safeMultiWriter{writers: writers}
+}
+
+// Write implements the io.Writer interface for safeMultiWriter.
+func (t *safeMultiWriter) Write(p []byte) (n int, err error) {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+
+	for _, w := range t.writers {
+		n, err = w.Write(p)
+		if err != nil {
+			return
+		}
+		if n != len(p) {
+			err = io.ErrShortWrite
+			return
+		}
+	}
+	return len(p), nil
+}
+
 func WithCombinedOutput(buf *bytes.Buffer) execOption {
 	return func(_ context.Context, c *exec.Cmd) error {
-		c.Stdout = io.MultiWriter(buf, c.Stdout)
-		c.Stderr = io.MultiWriter(buf, c.Stderr)
+		c.Stdout = newSafeMultiWriter(buf, c.Stdout)
```

Comment:
Does this prevent both stdout and stderr from writing to buffer at the same time? The locking only happens across a single `safeMultiWriter` object which does not prevent both stdout and stderr from writing to `buf` at the same time.

Reply:
Doh... the lock should be held on writing to the buffer, not the `io.MultiWriter`. Thanks for catching this.